src/main/scala/org/apache/spark/sql/redis/RedisSourceRelation.scala

```diff
@@ -142,8 +149,9 @@ class RedisSourceRelation(override val sqlContext: SQLContext,
           val conn = node.connect()
           foreachWithPipeline(conn, keys) { (pipeline, key) =>
             val row = rowsWithKey(key)
-            val encodedRow = persistence.encodeRow(keyName, row)
-            persistence.save(pipeline, key, encodedRow, ttl)
+            val encodedRow = persistence.encodeRow(keyName, row, ttlColumn)
+            val recordTTL = if (ttlColumn.isEmpty) ttl else row.getAs[Int](ttlColumn.get)
```

Comment:
We can rewrite it with `val recordTTL = ttlColumn.map(v => row.getAs[Int](v)).getOrElse(ttl)`. It looks to be more idiomatic approach.